Extensions/Leaderboards/leaderboardstools.ts

```diff
@@ -45,74 +116,196 @@ namespace gdjs {
       }
       _loaderContainer.appendChild(_loader);
 
-      export const setPlayerScore = function (
+      const getLastScoreSavingData = function ({
+        hasEnded,
+      }: {
+        hasEnded: boolean;
+      }): ScoreSavingData | null {
+        const dateFieldToUse = hasEnded
+          ? 'lastScoreSavingSucceededAt'
+          : 'lastScoreSavingStartedAt';
+        const scoreSavingDataArray = Object.values(
+          _savingDataByLeaderboard
+        ).filter((scoreSavingData) => !!scoreSavingData[dateFieldToUse]);
```

Comment:
This is a bit bypassing the type system, I would rather do something like:


```suggestion
        const scoreSavingDataArray = Object.values(
          _savingDataByLeaderboard
        ).filter((scoreSavingData) => !!(hasEnded
          ? scoreSavingData.lastScoreSavingSucceededAt
          : scoreSavingData.lastScoreSavingStartedAt));
```

(sure you evaluate the condition multiple times, but it's not less efficient than accessing a property by a string and at least TS will tell you when you rename a property that you must do it here)

Reply:
Or better for readibilty, use a function: `filter((scoreSavingData) => !!getDateField(scoreSavingData))`

with `getDateField = (scoreSavingData: ScoreSavingData) => hasEnded ? scoreSavingData.xxx : scoreSavingData.yyy)`.

You pay a function call which is has some overhead (callstack, jump, cache trashing), but we don't care about this (not performance sensitive code)